From: Etienne Millon <etienne.millon@gmail.com>
Date: Tue, 2 Nov 2010 17:04:53 +0100
Subject: replace_crc32

This patch converts the CRC32 global variable to RomCRC32, to avoid conflicts
with the CRC32 opcode.
---
 src/initc.c  |    6 +++---
 src/zmovie.c |    6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/initc.c b/src/initc.c
index 40348ec..40fba94 100644
--- a/src/initc.c
+++ b/src/initc.c
@@ -1651,7 +1651,7 @@ static void SPC7PathSetup(char *PathVar, const char *Default)
 
 extern unsigned int MsgCount, MessageOn;
 extern char *Msgptr;
-unsigned int SPC7110Entries, CRC32;
+unsigned int SPC7110Entries, RomCRC32;
 
 void SPC7PackIndexLoad()
 {
@@ -1798,9 +1798,9 @@ unsigned int showinfogui()
   // 42 is the answer, and the uCONSRT standard
 
   // calculate CRC32 for the whole ROM, or Add-on ROM only
-  CRC32 = (SplittedROM) ? crc32(0, ROM+addOnStart, addOnSize) : crc32(0, ROM, NumofBytes);
+  RomCRC32 = (SplittedROM) ? crc32(0, ROM+addOnStart, addOnSize) : crc32(0, ROM, NumofBytes);
   // place CRC32 on line
-  sprintf(CSStatus3+32, "%08X", CRC32);
+  sprintf(CSStatus3+32, "%08X", RomCRC32);
 
   i = (SplittedROM) ? infoloc + 0x1E + addOnStart: infoloc + 0x1E;
 
diff --git a/src/zmovie.c b/src/zmovie.c
index 4d8cc3f..683b1dc 100644
--- a/src/zmovie.c
+++ b/src/zmovie.c
@@ -75,7 +75,7 @@ typedef unsigned long long uint64;
 typedef unsigned __int64 uint64;
 #endif
 
-extern unsigned int versionNumber, CRC32, cur_zst_size, MsgCount, MessageOn;
+extern unsigned int versionNumber, RomCRC32, cur_zst_size, MsgCount, MessageOn;
 extern unsigned int JoyAOrig, JoyBOrig, JoyCOrig, JoyDOrig, JoyEOrig;
 extern unsigned char GUIReset, ReturnFromSPCStall, GUIQuit;
 extern unsigned char CMovieExt, mencoderExists, lameExists;
@@ -932,7 +932,7 @@ static bool zmv_create(char *filename)
     size_t filename_len = strlen(filename);
     strncpy(zmv_vars.header.magic, "ZMV", 3);
     zmv_vars.header.zsnes_version = versionNumber & 0xFFFF;
-    zmv_vars.header.rom_crc32 = CRC32;
+    zmv_vars.header.rom_crc32 = RomCRC32;
     zmv_vars.header.zst_size = cur_zst_size;
     zmv_vars.header.zmv_flag.start_method = (enum zmv_start_methods)MovieStartMethod;
     zmv_vars.header.zmv_flag.video_mode = romispal ? zmv_vm_pal : zmv_vm_ntsc;
@@ -1252,7 +1252,7 @@ static bool zmv_open(char *filename)
       Msgptr = "MOVIE STARTED.";
     }
 
-    if (zmv_vars.header.rom_crc32 != CRC32)
+    if (zmv_vars.header.rom_crc32 != RomCRC32)
     {
       static char buffer[29]; //"ROM MISMATCH. NEED: 01234567"
       sprintf(buffer, "ROM MISMATCH. NEED: %08X", zmv_vars.header.rom_crc32);
